File: Tests/src/test/scala/equellatests/pages/search/QuerySection.scala
package equellatests.pages.search

import com.tle.webtests.framework.PageContext
import equellatests.pages.{BrowserPage, WaitingBrowserPage}
import org.openqa.selenium.support.ui.ExpectedCondition
import org.openqa.selenium.{By, WebElement}

trait QuerySection extends BrowserPage {

  def queryField : WebElement = findElement(By.name("q"))

  def query : String = queryField.getAttribute("value")

  def query_=(q: String) : Unit = {
    queryField.clear()
    queryField.sendKeys(q)
  }

  def searchButton : WebElement = findElement(By.id("searchform-search"))
  def resultsUpdateExpectation : ExpectedCondition[_]
  def search(): this.type = {
    val expect = resultsUpdateExpectation
    searchButton.click()
    waitFor(expect)
    this
  }

}

File: autotest/Tests/src/main/java/com/tle/webtests/pageobject/searching/QuerySection.java
package com.tle.webtests.pageobject.searching;

import com.tle.webtests.framework.PageContext;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;

public class QuerySection extends AbstractQuerySection<QuerySection> {
  @FindBy(id = "searchform-search")
  private WebElement searchButton;

  public QuerySection(PageContext context) {
    super(context);
  }

  @Override
  protected WebElement getSearchButton() {
    return searchButton;
  }
}
